test/test_release.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+	"text/template"
+
+	c "github.com/flynn/flynn/Godeps/_workspace/src/github.com/flynn/go-check"
+	tc "github.com/flynn/flynn/test/cluster"
+)
+
+type ReleaseSuite struct {
+	Helper
+}
+
+var _ = c.Suite(&ReleaseSuite{})
+
+var releaseScript = bytes.NewReader([]byte(`
+export TUF_TARGETS_PASSPHRASE="flynn-test"
+export TUF_SNAPSHOT_PASSPHRASE="flynn-test"
+export TUF_TIMESTAMP_PASSPHRASE="flynn-test"
+
+export GOPATH=~/go
+src="${GOPATH}/src/github.com/flynn/flynn"
+
+export GIT_COMMIT=dev
+export GIT_BRANCH=dev
+export GIT_TAG="v20150131.0-test"
+export GIT_DIRTY=false
+
+# send all output to stderr so only version.json is output to stdout
+(
+  # tup hangs waiting on the FUSE socket after building the components, so
+  # for now we just kill the connection ater an appropriate amount of time
+  (sleep 30; sudo bash -c 'echo 1 > /sys/fs/fuse/connections/*/abort') &
```

Comment:
inotify would be good but I am not 100% sure what to wait for, and given this test takes ~4mins per iteration, I am tempted to leave as is for now and see how it holds up in CI.